rom/laghos_rom.cpp

```diff
@@ -4139,36 +4525,52 @@ void ROM_Operator::ForceIntegratorEQP(Vector & res) const
                 }
             }
         }
-
         loc_force *= eqpW[i] / ip.weight;  // Replace exact quadrature weight with EQP weight.
-
-        trial_fe->CalcShape(ip, shape);
-
-        Vector Vloc_force(loc_force.Data(), loc_force.NumRows() * loc_force.NumCols());
-
-        unitE.SetSize(shape.Size());
-        unitE = 1.0;
-
-        rhs.SetSize(h1dofs_cnt * dim);
-
-        v_e.SetSize(rhs.Size());
-
-        const int eos = elemIndex * nvdof;
-        for (int j=0; j<rdim; ++j)
-        {
-            for (int k=0; k<nvdof; ++k)
-                v_e[k] = W_elems(eos + k, j);
-
-            // Compute the inner product, on this element, with the j-th W vector.
-            res[j] += (v_e * Vloc_force) * (shape * unitE);
-        }  // Loop (j) over V basis vectors
+        
+		Vector Vloc_force(loc_force.Data(), loc_force.NumRows() * loc_force.NumCols());
+		Vector v_e(h1dofs_cnt * dim);
+
+		if (energy_conserve)
+		{
+			// energy-conserving EQP
+			for (int j = 0; j < rdim; ++j)
+			{
+				// TODO: v_e should be the jth velocity basis vector DOFs that
+				// correspond to the current element.
+				// Recall that the basis vectors have been renormalized by the
+				// induced Gram-Schmidt process, so that reduced Mv = I.
+
+				// I think the below is not right.
+				basis->GetBasisVectorV(false, j, v_e);	
```

Comment:
Here I want to set `v_e` to be the DOFs of the jth velocity basis vector corresponding to the current element, but I'm not sure how to do that. I started by getting the jth basis vector, but I believe I still need to reduce it to only the DOFs of the current element.

Reply:
In the `else` clause below, `v_e` is set. It looks like you just need to copy and paste that code.